lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2184,6 +2189,8 @@ void MemOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   }
 }
 
+std::optional<size_t> MemOp::getTargetResultIndex() { return 0; }
```

Comment:
I'm not sure if its result 0, this should be same behavior as `InstanceOp`

Reply:
Thanks for bringing this up, figuring out what the answers are and being explicit about it in a single place is part of the fun here! :smile: .

InstanceOp's have no "result" value corresponding to the symbol, and so field-specific symbols are rejected for that reason.  For memories, [this test](https://github.com/llvm/circt/blob/963f5e5518ab52164d3ebd1975c92461bd74d16f/test/Dialect/FIRRTL/errors.mlir#L922) looks to me to be specifically checking that a memory with symbols is considered to have fields (max of 0, presumably because of the empty bundle type)? (compare to the instance test)

Should we instead treat this as inner symbols do not target the result for these (and no per-field symbols)? Sounds good to me, but not the current behavior, consider the following example we currently accept:

```firrtl
firrtl.circuit "Foo" {
  firrtl.module @Foo() {
    %m = firrtl.mem sym [<@x,1,public>,<@y,2,public>] Undefined {depth = 16 : i64, name = "ReadMemory", portNames = ["read0"], readLatency = 1 : i32, writeLatency = 1 : i32} : !firrtl.bundle<addr: uint<4>, en: uint<1>, clk: clock, data flip: sint<8>>
  }
}
```

Reply:
Updated accordingly!  Presumably same for the CHIRRTL memory operations (cc #4874 -- they're not officially supported targets for inner symbols, and aren't verified presently), right? :+1:

Reply:
I think it doesn't make sense to allow per-field sym on memories. It becomes difficult to interpret it for multiple ports. So, no result should be good. If we need it in the future we can relax this.